x-pack/plugins/enterprise_search/public/applications/app_search/components/log_retention/log_retention_logic.ts

```diff
@@ -71,12 +69,10 @@ export const LogRetentionLogic = kea<MakeLogicType<LogRetentionValues, LogRetent
     ],
   }),
   listeners: ({ actions, values }) => ({
-    fetchLogRetention: async () => {
-      if (values.isLogRetentionUpdating) return; // Prevent duplicate calls to the API
+    fetchLogRetention: async (_, breakpoint) => {
+      await breakpoint(250); // Prevents duplicate calls to the API (e.g., when a tooltip & callout are on the same page)
```

Comment:
Just want to highlight this change / commit (https://github.com/elastic/kibana/pull/95227/commits/a111b6a2870bc17760d99aa470613bb3106fe563) - it's separate from the permissions fix but I kind of just jammed in here (apologies, feel free to yell at me and I can split it out!)

I realized while skimming this file that I could prevent duplicate API calls via Kea's `breakpoint` instead of bogarting the `isLogRetentionUpdating` state, and thought that felt cleaner & more Kea-y, so I made the change here.

This commit essentially completely reverts https://github.com/elastic/kibana/pull/88237/commits/c1978e5584424af383365269eb394444703cf33e (where I originally added the `isLogRetentionUpdating` workaround) and adds `jest.useFakeTimers`/`jest.runAllTimers` to test past the breakpoint.

Reply:
Oh this is a rad solution, I like this a lot. You may not even have to add a timeout, simply adding an empty `breakpoint()` (if that's possible) should work, I would imagine. That would have the benefit of not slowing down the experience.

Reply:
Gave that a shot! Looks like a 0 timeout isn't quite enough unfortunately because the tooltip and callout render separately on the page - there's enough distance or rendering time between them to trigger a duplicate call 😬

<img width="571" alt="" src="https://user-images.githubusercontent.com/549407/112339680-21c08d80-8c7d-11eb-9e01-082b5b3d2e28.png">
 
A timeout of `100` appears to dedupe but I'm 50/50 on leaving changing it vs leaving it at 250 🤔 The callouts aren't critical to the loading of the page and IMO it's okay that they render in a little later than usual, but definitely open to being argued otherwise on that and changing it to something faster!

Reply:
Actually you know what I totally forgot that `fetchLogRetention` is also used on the Settings page and is (currently) the only thing on that page, so it's definitely a nicer user experience to make those toggles interactable faster. I'll update this to `100`

```suggestion
      await breakpoint(100); // Prevents duplicate calls to the API (e.g., when a tooltip & callout are on the same page)
```